app/src/main/java/com/example/lutsak/rotation/MainActivity.java

```diff
@@ -1,13 +1,111 @@
 package com.example.lutsak.rotation;
 
-import android.support.v7.app.AppCompatActivity;
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public static final String LOG_TAG = MainActivity.class.getSimpleName();
+
+    private static int mCount = 0;
+    private Contact mContact;
+    private Button mCountButton;
+    private Button mSetNameButton;
+    private Button mShowNameButton;
+    private Button mGoWebButton;
+
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        Log.d(LOG_TAG, ">> onCreate");
+
+        mCountButton = findViewById(R.id.button1);
+        mSetNameButton = findViewById(R.id.setName);
+        mShowNameButton = findViewById(R.id.showName);
+        mGoWebButton = findViewById(R.id.button_web);
+        mGoWebButton.setOnClickListener(this);
+        mShowNameButton.setOnClickListener(this);
+        mCountButton.setOnClickListener(this);
+        mSetNameButton.setOnClickListener(this);
+
+        mContact = new Contact();
+
+        Log.d(LOG_TAG, "<< onCreate");
+    }
+
+    protected void onDestroy() {
+        super.onDestroy();
+        Log.d(LOG_TAG, "onDestroy");
+    }
+
+    protected void onPause() {
+        super.onPause();
+        Log.d(LOG_TAG, "onPause");
+    }
+
+    protected void onRestart() {
+        super.onRestart();
+        Log.d(LOG_TAG, "onRestart");
+    }
+
+    protected void onRestoreInstanceState(Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+        Log.d(LOG_TAG, ">> onRestoreInstanceState");
+
+        mContact = (Contact) getLastCustomNonConfigurationInstance();
```

Comment:
Where is this contact stored in this case?

Reply:
Is stored out of activity as object and using this method we need to downcast to Contact class.